builtin/types/karma/karma.proto

```diff
@@ -5,33 +5,27 @@ import "github.com/loomnetwork/go-loom/types/types.proto";
 package karma;
 
 message KarmaParams {
-    bool mutable_oracle = 1;
-    int64 max_karma = 2;
-    Address Oracle = 3;
-    repeated KarmaSourceReward sources = 4;
-    repeated Validator validators = 5;
-    int64 deploy_karma = 6;
-    repeated KarmaAddressSource users = 7;
+    KarmaConfig config = 1;
+    repeated KarmaAddressSource users = 2;
 }
 
-message KarmaParamsValidator {
-    KarmaParams Params = 1;
-    Validator Validator = 2;
-    Address Oracle = 3;
+message KarmaConfigValidator {
+    KarmaConfig config = 1;
+    Address oracle = 2;
 }
 
 message KarmaParamsValidatorNewMaxKarma {
     int64 max_karma = 1;
-    Validator Validator = 2;
+    Address oracle = 2;
 }
 
 message KarmaParamsValidatorNewOracle {
-    Address NewOracle = 1;
-    Validator Validator = 2;
+    Address new_oracle = 1;
+    Address old_oracle = 2;
```

Comment:
why do you need old oracle on an update? is this just being stored into the state db?

Reply:
The idea is that only the old/current Oracle can change the Oracle. I did it this way to reuse existing code with validator replaced by Oracle.

Reply:
On inspection, it looks like old oracles are kept. There is an error message that distinguishes between an old-oracle or an invalid one.